pkg/tsdb/prometheus/prometheus.go

```diff
@@ -163,9 +158,25 @@ func parseQuery(queries []*tsdb.Query, queryContext *tsdb.TsdbQuery) (*Prometheu
 		return nil, err
 	}
 
+	intervalFactor := queryModel.Model.Get("intervalFactor").MustInt64(1)
+
+	var step time.Duration
+	stepSeconds, err := queryModel.Model.Get("step").Int64()
+	if err != nil {
+		if err.Error() == "invalid value type" {
+			calculator := tsdb.NewIntervalCalculator(&tsdb.IntervalOptions{MinInterval: time.Second * 1})
```

Comment:
should this not look at the queryModel.Model.Get("interval")  (it contains the min interval value).

Reply:
We don't support that in the Prometheus data source. But it would be a good thing to align the min interval with Prometheus scrape interval. But that might be out of scope for this issue. 

The reason I introduced this option was to set a different lowest possible interval for Influxdb and Prometheus.

Reply:
We do support it an interval option (min step parameter in query editor)